EIC_superposed.Rmd

```diff
@@ -0,0 +1,137 @@
+    toc: true
+    number_sections: false
+    toc_float: true
+---
+
+```{r settings, echo = FALSE, results = "hide", message = FALSE}
+#' Define the path where we can find the mzML files:
+MZML_PATH <- "C:/Users/mgarciaaloy/Documents/mzML/"
+#if (!file.exists(MZML_PATH))
+#    stop("Can not find the directory with the mzML files: ", MZML_PATH)
+```
+
+# Preliminaries
+```{r, message=FALSE}
+study <- "standards_dilution" # specify "internal_standards" OR 
+                              #         "standards_dilution"
+mixnum <- 1 # specify which MIX
+polarity <- "POS" # specify "POS" or "NEG"
+
+library(xcms)
+library(reshape2)
+library(ggplot2)
+library(RColorBrewer)
+library(Rdisop)
+library(CompoundDb)
+
+# Set up parallel processing using 3 cores
+if (.Platform$OS.type == "unix") {
+    register(bpstart(MulticoreParam(2)))
+} else {
+    register(bpstart(SnowParam(2)))
+}
+```
+
+# Data import
+In the file `XXXX_files.txt` there is the information regarding 
+the injection sequence.
+```{r}
+injections <- read.table(paste0("data/", study, "_files.txt"), #import the file
+                         sep = "\t", header = TRUE, as.is = TRUE)
+myfiles <- injections$mzML # get file names
+myfiles <- myfiles[grep(polarity, myfiles)]# select files names of our polarity
+if(study == "standards_dilution"){
+  (myfiles <- myfiles[grep(paste0("MIX ", mixnum, "[A-Z]"), myfiles)])
+}
+```
+
+Get the information regarding the standards that are in the samples.
+```{r}
+std_info <- read.table(paste0("data/", study, ".txt"),
+                      sep = "\t", header = TRUE, as.is = TRUE)
+std_info$name <- c(substring(std_info$name, 1, 33))
+std_info <- std_info[!is.na(std_info[, grep(polarity, colnames(std_info))]),]
+if(study == "standards_dilution"){
+  std_info <- subset(std_info, mix == mixnum)
+}
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
+}
+write.csv(std_info, "x.csv", row.names = FALSE)
+std_info = read.csv("x.csv")
+std_info
+```
+
+# EICs
+```{r}
+da = 0.01
+RTd = 5
+if(study == "internal_standards"){
+  getPalette = colorRampPalette(brewer.pal(12, "Paired"))
+} else if(study == "standards_dilution"){
+  getPalette = colorRampPalette(brewer.pal(9, "YlOrRd"))
+}
+mycols = getPalette(length(myfiles))  
+data <- readMSData(paste0(MZML_PATH, myfiles), mode = "onDisk")
+
+for(i in 1:nrow(std_info)){
+  
+  mzvalue <- unlist(mass2mz(std_info$mzneut[i], adduct = as.character(std_info[i, grep(polarity, colnames(std_info))])))
+  
+  chr = chromatogram(data, 
+                     mz = c(mzvalue - da,  mzvalue + da),
+                     aggregationFun = "max")
+  
+  cmpname <- gsub("L-", "", std_info$name[i])
+  cmpname <- tolower(gsub(" .*", "", cmpname))
+  
+  if(study == "internal_standards"){ 
+    filename <- paste0("plots/IS_", cmpname, "_", 
```

Comment:
Suggestion: to be consistent with all other analyses we have, it might be better to save the plots into a folder *images/EIC_superposed* (that's the *standard* way how I name folders for analyses: *images/<name of Rmd creating the plots>*. Open for discussion.

Reply:
Eventually it might also be good to have a `dir.create("images/EIC_superposed", recursive = TRUE, showWarnings = FALSE)` at the beginning of the file to ensure that the directory is available.